### DISCUSSION TITLE
Manage spaCy pipeline package using pipenv with Pipfile / Pipfile.lock

### DISCUSSION THREAD

**Discussion by darioprencipe** [explosion/spaCy#12386] [Help: Installation] [answered]
Hello,

I am using `pipenv` to do package management in my python (Flask) dockerized application.

Need few components of your [trained pipeline](https://spacy.io/models/it) for Italian language. Now, in [your nice docs here](https://spacy.io/usage/models#download-pip), you outline how to install trained pipelines in automated processes using `pip`.

However, there's no mention on how to handle - in this case - `it_core_news_sm` as a package using `pipenv`, whereas my standard flow with any other packages would be:

1. install with pipenv (`pipenv install`)
2. update `Pipfile.lock` file
3. use `pipenv sync` at the end of my Dockerfile

What's the way to go about installing spacy pipelines using pipenv? 

Thanks a lot in advance!
Dario

**Comment by thomashacker** [comment, marked as answer]
Hey, thanks for your question.
A possible solution could be to directly [download the models from spacy-models](https://github.com/explosion/spacy-models/releases/tag/it_core_news_lg-3.5.0) and install them locally / or handle them however you'd handle local/private wheel packages.

Hope that helps!

**Reply by darioprencipe** [reply]
Hello,
what worked exactly as intended for me was running `pipenv install $(spacy info it_core_news_sm --url)`.

This added that package reference in my `Pipfile` such that I could lock it.

As less and less Python users use plain `pip` and `requirements.txt` to manage dependencies in their project, I think you should add a basic documentation support for pipenv usage. 

Thanks!

**Reply by thomashacker** [reply]
Hey,
Thanks for the info! We'll add this to our list
